src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/ListCaseNotificationHandler.java

```diff
@@ -65,7 +65,8 @@ public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCa
         return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
                && Arrays.asList(
                 Event.LIST_CASE,
-                Event.EDIT_CASE_LISTING
+                Event.EDIT_CASE_LISTING,
```

Comment:
I think it i is better to have it handled in a it's own event handler. And that would keep the handlers independent and to the functionality and easy to maintain in the future if there are any changes needed. Think about any refactoring to extract and re-use any common code.

Reply:
Created own handler and refactored to use common code. Had to extend as sonar was complaining on code duplication.